src/javascript/app/common/content_visibility.js

```diff
@@ -131,7 +141,10 @@ const ContentVisibility = (() => {
         else if (is_exclude !== rule_set_has_current) show_element = true;
         if (rule_set_has_eu_country && is_eu_country) show_element = !is_exclude;
 
-        if (mt5fin_rules.includes(mt5fin_company_shortcode)) show_element = !is_exclude;
+        // Check if list of mt5fin_company_shortcodes is array type and filter with defined mt5fin rules
+        if (Array.isArray(mt5fin_company_shortcodes)) {
+            if (mt5fin_company_shortcodes.some(el => mt5fin_rules.includes(el))) show_element = !is_exclude;
+        }
```

Comment:
should we handle else as before?

Reply:
Don't think we need `else` in this case as if there's no `array` found, the rules won't be applied and content visibility will ignore the `data-show` rules set in the templates.